### PR TITLE
[#145788259] Exclude occupancies from facility orders view

### PR DESCRIPTION
**Pull request by jhanggi**
This creates a new `NewInprocessController` concern that is used for the sorting and loading of the `index` view. It also updates the `non_reservation` scope to be `plain_orders` (I couldn't come up with a better name) that filters by the product type.

### CODE REVIEW

File: vendor/engines/secure_rooms/spec/controllers/facility_orders_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  let!(:problem_occupancy) { create(:occupancy, :problem_with_order_detail, secure_room: secure_room, user: facility_director, account: account) }
+
+  let(:facility_director) { create(:user, :facility_director, facility: facility) }
+
+  before { sign_in facility_director }
+
+  describe "index" do
+    before { get :index, facility_id: facility }
+
+    it "does not include te occupancies" do
```
Comment (waggles) [inline vendor/engines/secure_rooms/spec/controllers/facility_orders_controller_spec.rb]: `te`

File: vendor/engines/secure_rooms/spec/controllers/facility_orders_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      expect(response).to be_success
+      expect(assigns(:order_details)).to eq([])
+    end
+  end
+
+  describe "show_problems" do
+    before { get :show_problems, facility_id: facility }
+
+    it "does not include te occupancies" do
```
Comment (waggles) [inline vendor/engines/secure_rooms/spec/controllers/facility_orders_controller_spec.rb]: oh here too I guess

File: vendor/engines/secure_rooms/app/controllers/secure_rooms/facility_occupancies_controller.rb [hunk 3]
```diff
@@ -2,32 +2,26 @@ module SecureRooms
...
       @active_tab = "admin_occupancies"
     end
 
-    def index
-      @order_details = new_or_in_process_orders.order(@order_by_clause).paginate(page: params[:page])
-    end
+    # GET /facilities/:facility_id/occupancies
+    # Provided by NewInprocessController
+    # def index
+    # end
```
Comment (waggles) [inline vendor/engines/secure_rooms/app/controllers/secure_rooms/facility_occupancies_controller.rb]: ahh this is a great idea

File: vendor/engines/secure_rooms/spec/controllers/facility_orders_controller_spec.rb (hunk 1 above)
Comment (jhanggi) [inline vendor/engines/secure_rooms/spec/controllers/facility_orders_controller_spec.rb]: Sorry, should be `teh`.

Comment (waggles) [inline vendor/engines/secure_rooms/spec/controllers/facility_orders_controller_spec.rb]: or even "t3h"

File: app/models/order_detail.rb [hunk 4]
```diff
@@ -260,8 +255,10 @@ def can_be_viewed_by?(user)
       .where.not(statement_id: nil)
   }
 
-  scope :non_reservations, -> { joins(:product).where("products.type <> 'Instrument'") }
-  scope :reservations, -> { joins(:product).where("products.type = 'Instrument'") }
+  # Supports single type or an array of product types
+  scope :for_product_type, ->(product_type) { joins(:product).where(products: { type: product_type }) }
+  scope :plain_orders, -> { for_product_type(["Item", "Service"]) }
```
Comment (meara) [inline app/models/order_detail.rb]: `untimed_orders`, maybe? To contrast with `time_data`? Or `timeless_orders` lol.

File: vendor/engines/secure_rooms/app/controllers/secure_rooms/facility_occupancies_controller.rb (hunk 3 above)
Comment (meara) [inline vendor/engines/secure_rooms/app/controllers/secure_rooms/facility_occupancies_controller.rb]: Yeah I like this, nice solution.

File: app/models/order_detail.rb (hunk 4 above)
Comment (jhanggi) [inline app/models/order_detail.rb]: `untimed_orders` is a decent name, but it seems like with that name they've become the special case. Especially when the front end just calls them "Orders". Naming is hard.

**Comment by jhanggi** [comment]
Updated. Ended up going with `untimed_orders`.

**Pull request merged**